Document the CriticalCSS/DiscountFlashPrevention plugins

## 🗄️ Docs report

Possibly two of the coolest plugins we've built in a while, and we haven't documented it.

Hi @aaronfrost, I'm getting this error.

```
Cannot find module '@scullyio/scully-plugin-docs-link-update'
```

Am I missing something or I just have to install (globally?) the missing dependency?

![image](https://user-images.githubusercontent.com/4236211/100699860-142b8300-33c2-11eb-8b45-9183aabbc7ed.png)

--- a/docs/Reference/plugins/built-in-plugins/criticalcss.md
+++ b/docs/Reference/plugins/built-in-plugins/criticalcss.md
@@ -1,0 +1,18 @@
+---
+title: Critical CSS Plugin
+published: true
+lang: en
+position: 100
+---
+
+# `critical-css` Plugin
+
+## Overview
+
+Scully uses this plugin to inline the critical above the fold CSS into the HTML, and then **lazy-load** the CSS that is needed for the rest of the page. This will remove CSS from being blocking. It will be ready before your SPA will boot.
+
+## Install
+
+```
+npm install -D @scullyio/scully-plugin-critical-css
+```

--- a/docs/Reference/plugins/built-in-plugins/criticalcss.md
+++ b/docs/Reference/plugins/built-in-plugins/criticalcss.md
@@ -1,5 +1,5 @@
 ---
-title: Critical CSS Plugin
+title: critical css Plugin
 published: true
 lang: en
 position: 100
@@ -9,10 +9,83 @@ position: 100
 
 ## Overview
 
-Scully uses this plugin to inline the critical above the fold CSS into the HTML, and then **lazy-load** the CSS that is needed for the rest of the page. This will remove CSS from being blocking. It will be ready before your SPA will boot.
+Scully uses this plugin to inline the critical above the fold CSS into the HTML, and then **lazy-load** the CSS that is needed for the rest of the page.This will remove CSS from being blocking. It will be ready before your SPA will boot.
 
 ## Install
 
 ```
 npm install -D @scullyio/scully-plugin-critical-css
 ```
+
+## Usage
+
+Add it to your `scully.<projectname>.config.ts` like this:
+
+```typescript
+import { criticalCSS } from '@scullyio/scully-plugin-critical-css';
+
+/** the seoHrefOptimise is optional **/
+const defaultPostRenderers = ['seoHrefOptimise', criticalCSS];
+
+export const config: ScullyConfig = {
+  /** more config here */
+  defaultPostRenderers,
+  /** more config here */
+  routes: {
+    /** more config here */
+  },
+};
+```
+
+The above config will use the plugin on all routes. If you want to use in on a single route, add it to the config of that particular route like this:
+
+```typescript
+export const config: ScullyConfig = {
+  /** more config here */
+  routes: {
+    someRoute: {
+      type: 'contentFolder', // Or any other type
+      postRenderers: = [criticalCSS],
+    },
+     /** more route configs here */
+  }
+}
+```
+
+## Settings
+
+You can configure this plugin by using the setPluginConfig helper like this:
+
+```typescript
+setPluginConfig(criticalCSS, {
+  inlineImages: false,
+});
+```
+
+The plugin had the following settings:
+
+```typescript
+export interface CriticalCSSSettings {
+  /** inline images into the pages when smaller then 10240 bytes */
+  inlineImages?: boolean;
+  /** Width of the target viewport */
+  width?: number;
+  /** Height of the target viewport */
+  height?: number;
+  /** An array of objects containing height and width. Takes precedence over width and height if set */
+  dimensions?: {
+    width: number;
+    height: number;
+  }[];
+  /** An array with fully qualified paths to assets, if none is given, the root, and the root/assets will be used to look for static assets*/
+  assets?: string[];
+  /** Ignore some css rules */
+  ignore?: {
+    atrule?: string[];
+    rule?: string[];
+    decl?: (node, value) => boolean;
+  };
+}
+```
+
+For details see the [critical tool page](https://github.com/addyosmani/critical)

--- a/docs/Reference/plugins/built-in-plugins/flashPrevention.md
+++ b/docs/Reference/plugins/built-in-plugins/flashPrevention.md
@@ -1,0 +1,102 @@
+---
+title: flash prevention Plugin
+published: true
+lang: en
+position: 100
+---
+
+# `flash-prevention` Plugin
+
+## Overview
+
+The `scully-plugin-flash-prevention` is a postRenderer that helps you hide any flashes that your app may be experiencing once you add Scully to your project.
+
+After adding Scully, your app will appear instantly because the pre-rendered HTML and CSS is immediately available. After appearing instantly, the JavaScript and CSS files will download and then your Angular app will bootstrap (init). When it bootstraps, the pre-rendered version may disappear for a moment, and then once the app is ready, the view will re-appear. This disappearing-then-appearing is very normal for apps that are pre-rendered on a server. This plugin is to prevent that.
+
+This plugin shows the pre-rendered copy of your app until your app is fully render and the flash is over. It then shows your app and deletes the copy.
+
+## How it works
+
+Before this plugin, you app would pre-render and then save to file, like this:
+
+```html
+<app-root _nghost-abc="" ng-version="9.0.1" class="my-class">
+  // The entire content of your app here
+</app-root>
+```
+
+After this plugin, you will see the following in your pre-rendered template:
+
+```html
+<app-root class="my-class"></app-root>
+<app-root-scully _nghost-abc="" ng-version="9.0.1" class="my-class">
+  // The entire content of your app here
+</app-root-scully>
+```
+
+This app-root-scully will be the pre-rendered copy of your app. Prior to your app being rendered fully, app-root will be hidden and app-root-scully will be displayed.
+
+Once your app has fully bootstrapped, app-root-scully will be hidden and then 100ms later removed from the DOM. The mechanism that shows and hides these two is CSS. There is some CSS added during the Scully build that looks like the following:
+
+```css
+body:not(.loaded) app-root {
+  display: none;
+}
+body.loaded app-root-scully {
+  display: inherit;
+}
+```
+
+Once the app has been fully loaded, the `loaded` class is added to the `<body>` tag.
+
+And that's how it all works!!!
+
+# Getting Started
+
+## Install
+
+```
+npm install -D @scullyio/scully-plugin-flash-prevention
+```
+
+## Usage
+
+Add the postRenderer to your scully.config:
+
+```typescript
+// Add this line to your imports
+const { getFlashPreventionPlugin } = require('scully-plugin-flash-prevention');
+
+// Add the following to your `scully.config.postRenderers`
+exports.config = {
+  ...
+  postRenderers : [getFlashPreventionPlugin({appRootSelector: 'custom-app-root'})],
+  ...
+}
+```
+
+You only need to pass the `{appRootSelector: 'custom-app-root'}` if your app has a selector other than `app-root`. It is defaulted to `app-root`.
+
+Update `app.module` to include `alwaysMonitor` in the `ScullyLibModule.forRoot` call.
+
+```typescript
+ScullyLibModule.forRoot({
+  useTransferState: true,
+  alwaysMonitor: true,  <-- Add this line to your `app.module.ts`
+});
+```
+
+Apply any styles from `app-root` to `app-root-scully` as well. Any styles that are in your `app.component.(css|scss|less)` need to be applied to the copy of your app that was made. This means that you need to possibly move any styles that apply to the `app-root` specifically, and put them in a location where you can also make those styles apply to `app-root-scully` as well. See here:
+
+```css
+// BEFORE
+app-root {
+  ... some styles;
+}
+
+// AFTER
+app-root,
+app-root-scully {
+  ... some styles;
+}
+```

--- a/libs/plugins/scully-plugin-critical-css/README.md
+++ b/libs/plugins/scully-plugin-critical-css/README.md
@@ -2,7 +2,7 @@
 
 > BETA notice: this plugin is still in beta
 
-- Description: With this Scully render plugin, we will inline the critical above the fold CSS into the HTML, and then "lazy-load" the CSS that is needed for the rest of the page. It will be ready before your SPA will boot. This will remmove CSS from being blocking.
+- Description: With this Scully render plugin, we will inline the critical above the fold CSS into the HTML, and then "lazy-load" the CSS that is needed for the rest of the page. It will be ready before your SPA will boot. This will remove CSS from being blocking.
 - Type: Render Plugin
 
 This plugins uses the [critical](https://github.com/addyosmani/critical#critical) tool that is maintained by google, and is listed as the official endorsed option on [web.dev](https://web.dev/extract-critical-css/)
@@ -31,7 +31,8 @@ export const config: ScullyConfig = {
   /** more config here */
   routes: {
     /** more config here */
-  }
+  },
+};
 ```
 
 The above config will use the plugin on _all_ routes. If you want to use in on a single route, add it to the config of that particular route like this:

